HealthcareSystemFrontends/GraphicalEditor/MainWindow.xaml.cs

```diff
@@ -686,37 +686,86 @@ private void ShowMedicineSearchResultObjectOnMapButton_Click(object sender, Rout
+            int patientCardId = ((PatientBasicDTO)AppointmentSearchPatientComboBox.SelectedItem).PatientCardId;
+
+            int doctorSpecialtyId = ((SpecialtyDTO)AppointmentDoctorSpecializationComboBox.SelectedItem).Id;
+
+            SearchPriority searchPriority;
+            if ((bool)AppointmentSearchDoctorPriorityRadioButton.IsChecked)
+            {
+                searchPriority = SearchPriority.Doctor;
+            }
+            else
+            {
+                searchPriority = SearchPriority.Date;
+            }
+         
+
+            List<int> appointmentRequiredEquipmentTypes = new List<int>();
+            foreach (EquipmentTypeForViewDTO equipmentType in AllEquipmentTypes)
+            {
+                if(equipmentType.IsSelected)
+                {
+                    appointmentRequiredEquipmentTypes.Add(equipmentType.EquipmentType.Id);
+                }
+            }
+
+            AppointmentService appointmentService = new AppointmentService();
+            AppointmentSearchWithPrioritiesDTO appointmentSearchParametersDTO = new AppointmentSearchWithPrioritiesDTO(new BasicAppointmentSearchDTO(patientCardId, doctorJmbg, appointmentRequiredEquipmentTypes, new DateTime(appointmentSearchStartDateTime.Ticks, DateTimeKind.Utc), new DateTime(appointmentSearchEndDateTime.Ticks, DateTimeKind.Utc)), searchPriority, doctorSpecialtyId);
+            List<ExaminationDTO> examinationDTOs =  appointmentService.GetFreeAppointments(appointmentSearchParametersDTO);
+            ExaminationSearchResults = examinationDTOs;
+            List<ExaminationDTO>  examinationDTOsWithoutDuplicates = RemoveAppointmentsWithDuplicateTimes(examinationDTOs);
+            
+            AppointmentSearchResultsDataGrid.ItemsSource = examinationDTOsWithoutDuplicates;
+
             AppointmentSearchScrollViewer.ScrollToBottom();
         }
 
         private void AppointmentSearchResultsDataGrid_SelectionChanged(object sender, SelectionChangedEventArgs e)
         {
+            ExaminationDTO selectedExaminationAppointment = (ExaminationDTO)AppointmentSearchResultsDataGrid.SelectedItem;
 
+            SelectedAppointmentRoomComboBox.ItemsSource =  GetFreeRoomsByAppointment(ExaminationSearchResults, selectedExaminationAppointment);
         }
 
         private void SelectedAppointmentRoomComboBox_SelectionChanged(object sender, SelectionChangedEventArgs e)
         {
-
+            if(SelectedAppointmentRoomComboBox.SelectedItem != null)
+            {
+                int selectedAppointmentRoomId = (int)SelectedAppointmentRoomComboBox.SelectedItem;
+                MapObject selectedAppointmentRoomMapObject = _mapObjectController.GetMapObjectById(selectedAppointmentRoomId);
+                ShowSelectedSearchResultObjectOnMap(selectedAppointmentRoomMapObject);
+            }
         }
 
         private void ScheduleAppointmentButton_Click(object sender, RoutedEventArgs e)
         {
-
+           
         }
 
```

Comment:
Why is this left empty?
Will it be implemented in near future?

Reply:
Yes, it is going to be implemented in the next PR. I will just paste code here from the old local repo.